include/flatbuffers/flexbuffers.h

```diff
@@ -384,7 +384,7 @@ class Reference {
 
   Reference(const uint8_t *data, uint8_t parent_width, uint8_t packed_type)
       : data_(data), parent_width_(parent_width) {
-    byte_width_ = 1U << static_cast<BitWidth>(packed_type & 3);
+    byte_width_ = static_cast<BitWidth>(1U << (packed_type & 3));
```

Comment:
byte_width_  is not a BitWidth type. This cast is weird.
And also could you please put those initialization like

```
  Reference(const uint8_t *data, uint8_t parent_width, uint8_t packed_type)
      : data_(data),
        parent_width_(parent_width),
        byte_width_(1U << static_cast<BitWidth>(packed_type & 3)),
        type_(static_cast<Type>(packed_type >> 2))
```

Reply:
The better way was  `1 << packed_type & 3;`

Reply:
Fixed in by [0e79e56](https://github.com/google/flatbuffers/commit/0e79e56427964bb20130eb393a8ce321d94bad76)

Reply:
yeah, enum-class's code is right, is more clearly.

But 1 << (packed_type & 3) is int, also has warning as error.
flexbuffers.h:362:21: error: implicit conversion loses integer precision: 'int' to 'uint8_t' (aka 'unsigned char') [-Werror,-Wimplicit-int-conversion]
      byte_width_(1 << (packed_type & 3)) {
                 ~~~^~~~~~~~~~~~~~~~~~~~
<img width="1929" alt="image" src="https://user-images.githubusercontent.com/22816236/206152966-06bca60f-e48f-47f8-b4c2-4fa12f713b41.png">

byte_width_(static_cast<uint8_t>(1 << (packed_type & 3))) is better, fix implicit-int-conversion warning.
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/22816236/206153298-8ae4a211-70de-4f6a-ade9-4ae1296f3107.png">

Reply:
It also has clang format problem :))

Reply:
In our project, we add cxx flags -Wall -Werror -Wextra -Wimplicit-int-conversion and we use clang compiler(android/ios/mac/linux).
int to uint8_t will treat warning as error, because of implict conversion loses interger precision.

Why we add cxx flags -Wimplicit-int-conversion, because of some case, for example:
void func(int32_t value) {
  std::map<std::string, std::string> my_map;
  my_map["key"] = value; // this line will make logic error, becase value implict convert to ascii char.
}

so we need add -Wimplicit-int-conversion to forbid implicit conversion, and our project use flatbuffer, we want fix this warning.

Reply:
@runhwguo I have prepared a p.r.